### ISSUE TITLE
Compatibility with ExLlamaV2

### ISSUE THREAD

**Issue by accountForIssues** [noamgat/lm-format-enforcer#21]
Thanks so much for sharing your project. I could get it working easily with transformers.

I have a few questions.

1. Now that your PR has been merged with vLLM (version not yet released), does the example still work ? Do we still need to patch the functions ? I got an error, "Too many values to unpack" during the unpacking penalties step. I followed the notebook to the word.

2. Can support be added for [ExLlamaV2](https://github.com/turboderp/exllamav2/) ? The library is extremely high performing and being able to use 70b quants at 7 to 13b speeds would be amazing.

Let me know if I should create separate issues. Thanks again.

**Comment by noamgat** [comment]
Regarding vLLM, I just ran the sample notebook and it worked correctly. I would double check your integration.
We still need to patch the functions, until the next vLLM release. When it will be ready, I will add an official vllm integration to the "integrations" module and simplify the sample notebook accordingly.

Regarding ExLlamaV2 - from a brief glance into the API it looks possible, as the sampler settings has the filters array, which would allow to set up a filter similar to how the huggingface integration works. However I don't currently have time to develop it, and the API does not seem that well documented, so I won't get to it in the coming days. PRs welcome :)

Otherwise, lets dedicate this post to ExLlamaV2, and see how many votes it gets.

**Comment by jpeig** [comment]
Would also vote to include ExLlamaV2. Its basically a 5x speed improvement - supercharging json output.

**Comment by jpeig** [comment]
Instead of ExLlamaV2, it may be an even better idea to provide support for https://github.com/huggingface/text-generation-inference.

It's similar to vLLM and includes support for ExLlamaV2, including many other features.

**Comment by noamgat** [comment]
Small update: lm-format-enforcer 0.6.5 with an official integration layer, now that vLLM released a version with the logits processor API.
The [vLLM sample notebook](https://github.com/noamgat/lm-format-enforcer/blob/main/samples/colab_vllm_integration.ipynb) has been updated, it is much simpler and more powerful now (includes regex and analysis examples).

**Comment by jpeig** [comment]
@noamgat 

Quite some significant changes - amazing. However, I created my own fork of the vLLM API server and hardcoded your custom sampler. Would you recommend switching it with the logits processor implementation? Would there be significant speed or accuracy gains?

**Comment by noamgat** [comment]
I would switch to the mainline version so you don't have a fork to
maintain... Performance / quality wise they should be equivalent

**Comment by noamgat** [comment]
As of v0.7.1, ExLlamaV2 is supported!

If you want to request support for text-generation-inference, please create a new issue, and also comment and thumbsup on this ticket so we can get the needed API support there: https://github.com/huggingface/text-generation-inference/issues/1269
